utils/atomic-data-publisher/src/database.rs

```diff
@@ -0,0 +1,1070 @@
+use std::sync::Arc;
+
+use anyhow::Result;
+use chrono::{Duration, Utc};
+use serde::{Deserialize, Serialize};
+use sqlx::{postgres::PgPoolOptions, PgPool, Row};
+use tracing::{debug, info, warn};
+
+use crate::{
+  config::{DatabaseConfig, PollingJob},
+  errors::AtomicDataError,
+  metrics::MetricsCollector,
+};
+
+const MIN_CHUNK_SIZE: u64 = 50_000;
+const MAX_CHUNK_SIZE: u64 = 500_000;
+const DEFAULT_CHUNK_PERCENTAGE: f64 = 0.25;
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct ChangeRecord {
+  pub job_name: String,
+  pub query_name: String,
+  pub target_block: u64,
+  pub atomic_data: serde_json::Value,
```

Comment:
There seems to be a lot of validation that this value is an array. 
Could we reduce almost all of that by changing this field to a `Vec<serde_json::Value>`?